### PR TITLE
JSON: Use OrderedDict to maintain unit order

### LINKED ISSUE #3394: json2po output strings in wrong order
Running `json2po` outputs a valid PO files, but the strings are in the wrong order. I’m not sure exactly *which* order it is, but is *neither* the original order, the reverse order or alphabetically. Here’s an example JSON file:

```json
    {
        "foo": "foo",
        "bar": "bar",
        "baz": "baz"
    }
```

Running `json2po` on this file results in the following (with a PO file header):

```po
    #: .baz
    msgid "baz"
    msgstr ""
    
    #: .foo
    msgid "foo"
    msgstr ""
    
    #: .bar
    msgid "bar"
    msgstr ""
```

Having the strings in a different order than in the original file makes it much harder to correctly translate the file, as one loses context.

I observe this bug both in version 1.13.0 and 1.14.0-rc1.

### PR DESCRIPTION
**Pull request by dwaynebailey**
By default JSON will use a normal dict for objects.  Overridding this
and using OrderedDict means we can maintain order and thus translator
context.

Thanks http://stackoverflow.com/a/6921760/2167584

Fixes #3394

### CODE REVIEW

File: requirements/required.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # standard library in Python >= 2.7
 argparse
 six>=1.9
+ordereddict ;python_version <= '2.6'       # JSON
```
Comment (unho) [inline requirements/required.txt]: I would remove the JSON comment. This package might be necessary to address #3396 in a proper way.

Comment (dwaynebailey) [inline requirements/required.txt]: That can happen when that issue is addressed.

Comment (unho) [inline requirements/required.txt]: I am ok with that.

File: translate/storage/jsonl10n.py [hunk 2]
```diff
@@ -70,6 +70,11 @@
 
 import json
 import os
+try:
+    from collections import OrderedDict
+except ImportError:
+    # Python 2.6 does not have OrdereDict and also can't use it in json.loads()
```
Comment (unho) [inline translate/storage/jsonl10n.py]: Typo: OrderedDict

Comment (unho) [inline translate/storage/jsonl10n.py]: Also json.loads() can't use OrderedDict in 2.6, even if it were available.

File: setup.py [hunk 3]
```diff
@@ -398,8 +398,9 @@ def parse_requirements(file_name):
     requirements = []
     with open(file_name, 'r') as fh:
         for line in fh:
-            # Ignore comments, blank lines and included requirements files
-            if re.match(r'(\s*#)|(\s*$)|(-r .*$)', line):
+            # Ignore comments, blank lines, included requirements files,
+            # extended requirements syntax
+            if re.match(r'^(\s*#|\s*$|-r .*$|.*;)', line):
```
Comment (unho) [inline setup.py]: Is this change related to the issue. If not please put in separate commit.

Comment (dwaynebailey) [inline setup.py]: Nah I can lose this now

**Comment by unho** [comment]
lgtm if Travis passes.

**Comment by phlax** [comment]
lgtm
